### PR TITLE
Emit anonymous delegates, used for dynamic, in deterministic order

### LINKED ISSUE #1440: Ordering of synthesized delegates in metadata should be deterministic
Ordering of synthesized delegates in metadata should be deterministic. See SynthesizedDelegateMetadataOrder test.

### PR DESCRIPTION
**Pull request by gafter**
Fixes #1440 
@tmat @AlekseyTs @agocke @VSadov

### CODE REVIEW

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs [hunk 1]
```diff
@@ -383,7 +383,17 @@ private void GetCreatedSynthesizedDelegates(ArrayBuilder<SynthesizedDelegateSymb
...
                     }
                 }
-                // Should be sorted, same as AnonymousTypeTemplates. See VB.
+                builder.Sort(SynthesizedDelegateSymbolComparer.Instance);
+            }
+        }
+
+        private class SynthesizedDelegateSymbolComparer : IComparer<SynthesizedDelegateSymbol>
+        {
+            public static SynthesizedDelegateSymbolComparer Instance = new SynthesizedDelegateSymbolComparer();
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs]: readonly?

**Comment by AlekseyTs** [comment]
LGTM (see a comment about "readonly")

**Comment by agocke** [comment]
LGTM

**Pull request merged**